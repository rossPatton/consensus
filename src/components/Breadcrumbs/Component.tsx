import React, { memo } from 'react';
import { Link } from 'react-router-dom';

export const BreadcrumbsComponent = memo((props: any) => (
  <ul className="fs6 fw600 mB2 lh1 lsNone fx fxWrap brdB1 mB3 pB3">
    {console.log('props => ', props)}
    {props.country && (
      <li className="mR1">
        {!props.region && props.country.name}
        {props.region && (
          <Link to={`/${props.country.code}`}>
            {props.country.name}
          </Link>
        )}
      </li>
    )}
    {props.region && (
      <li className="mHide mR1">/</li>
    )}
    {props.region && (
      <li className="mR1">
        {!props.city && props.country.name}
        {props.city && (
          <Link to={`/${props.country.code}/${props.region.code}`}>
            {props.region.name}
          </Link>
        )}
      </li>
    )}
    <li className="mHide mR1">/</li>
    <li className="mR1">
      <Link to="/us/ny/new-york-city">
        New York City
      </Link>
    </li>
  </ul>
));
