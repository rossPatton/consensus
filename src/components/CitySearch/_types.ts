export type tEventTypes = React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>;
type tUpdateState = (key: any, value: string | number | object | boolean) => void;

// these are inherited by the form
type tProps = {
  city: string,
  cityId: number | null,
  geo: tGeo,
  region: string,
  regionId: number | null,
  session: tSession,
};

export type tContainerProps = tProps & {
  citiesThunk: tThunk<tCity[]>,
  getCitiesDispatch: (query?: {region: string}) => tThunkPayload<tCity[]>,
  // inherited. updates form state
  updateState: tUpdateState,
};

export type tState = {showRegionField: boolean};
export type tKeyUnion = 'city' | 'cityId' | 'region';

export type tComponentProps = tState & tProps & {
  cities: tCity[],
  // local update state. just updates the city and region inputs
  updateState: (state: Partial<tProps>) => void,
  toggleRegionField: () => void,
};

export type tStore = {
  cities: tThunk<tCity[]>,
  geo: tThunk<tGeo>,
  session: tThunk<tSession>,
};
