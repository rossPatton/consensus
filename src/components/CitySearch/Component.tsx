import _ from 'lodash';
import React, {memo} from 'react';

import stateCodeMap from '../../json/usa/stateCodeMap.json';
import {tComponentProps} from './_types';

export const CitySearchComponent = memo((props: tComponentProps) => {
  const onCityChange = (ev: React.ChangeEvent<HTMLSelectElement>) => {
    const city = _.find(props.cities, c => c.name === ev.currentTarget.value);
    if (city) {
      props.updateState({
        city: city.name,
        cityId: city.id,
        region: city.region,
        regionId: city.regionId,
      });
    }
  };

  const onRegionChange = (ev: React.ChangeEvent<HTMLSelectElement>) => {
    const region = ev.currentTarget.value;
    if (region !== props.region) {
      props.updateState({
        city: '',
        cityId: 0,
        region,
        regionId: 0,
      });
    }
  };

  return (
    <>
      {props.showRegionField && (
        <>
          <h2 className="fs5 mB1 lh1">
            Pick a different state
          </h2>
          <label htmlFor="stateSelect">
            <select
              name="type"
              id="stateSelect"
              className="mB4 row bgWhite"
              value={props.region}
              onBlur={onRegionChange}
              onChange={onRegionChange}>
              {Object.keys(stateCodeMap).map(state => (
                <option key={state} value={state}>
                  {state}
                </option>
              ))}
            </select>
          </label>
        </>
      )}
      <h2 className="fs5 mB1 lh1">
        City in <span className="dInBl mR2">
          {(!props.showRegionField && props.geo.region) && props.geo.region}
          {(props.showRegionField && props.region) && props.region}
        </span>
        {!props.showRegionField && (
          <button
            type="button"
            onClick={props.toggleRegionField}>
            Not the right state?
          </button>
        )}
      </h2>
      {props.label && (
        <p className="fs5 copyBlack mB1">
          {props.label}
        </p>
      )}
      <div className="mB4">
        {props.cityId === 0 && (
          <label htmlFor="citySelect">
            <select
              name="type"
              id="citySelect"
              className="row bgWhite"
              value={props.city}
              onBlur={onCityChange}
              onChange={onCityChange}>
              {props.cities.map((city: tCity) => (
                <option key={city.id} value={city.name}>
                  {city.name}, {city.region}
                </option>
              ))}
            </select>
          </label>
        )}
        {props.cityId > 0 && (
          <div className="brdA1 p3 black br8 dInBl">
            <b>{props.city}</b>, <span className="dInBl mR3">{props.region}</span>
            <button
              type="button"
              onClick={() => {
                props.updateState({
                  city: '',
                  cityId: 0,
                });
              }}>
              Not the right city?
            </button>
          </div>
        )}
      </div>
    </>
  );
});
