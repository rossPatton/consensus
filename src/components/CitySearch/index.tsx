import _ from 'lodash';
import React, {PureComponent} from 'react';
import {connect} from 'react-redux';

import {GenericLoader} from '~app/containers';
import {getCities} from '~app/redux';

import {tContainerProps, tStore} from './_types';
import {CitySearchComponent} from './Component';

// use within an existing form, tie to it's updateState method
class CitySearchContainer extends PureComponent<tContainerProps> {
  static defaultProps = {
    city: 'New York',
    region: 'New York',
    showResetButton: true,
  };

  constructor(props: tContainerProps) {
    super(props);
    let region = 'New York';
    const sessionRegion = _.get(props, 'session.profile.region', null);
    const geoRegion = _.get(props, 'geo.region', null);

    if (typeof sessionRegion === 'string' && sessionRegion !== '') {
      region = sessionRegion;
    } else if (typeof geoRegion === 'string' && geoRegion !== '') {
      region = geoRegion;
    }

    props.getCitiesDispatch({ region });
  }

  updateState = async (state: {[key: string]: unknown}) => {
    if (state.region) {
      await this.props.getCitiesDispatch({region: state.region as string});
    }

    // set local state for the city input, but then pass it up to the form
    this.props.updateState(null, state);
  }

  render() {
    return (
      <GenericLoader
        isLoading={this.props.citiesThunk.isLoading}
        render={() => (
          <CitySearchComponent
            {...this.props}
            cities={this.props.citiesThunk.data}
            geo={this.props.geo}
            session={this.props.session}
            updateState={this.updateState}
          />
        )}
      />
    );
  }
}

const mapStateToProps = (store: tStore) => ({
  citiesThunk: store.cities,
  geo: store.geo.data,
  session: store.session.data,
});

const mapDispatchToProps = (dispatch: Function) => ({
  getCitiesDispatch: (query: {region: string}) => dispatch(getCities(query)),
});

const CitySearch = connect(
  mapStateToProps,
  mapDispatchToProps,
)(CitySearchContainer);

export default CitySearch;
