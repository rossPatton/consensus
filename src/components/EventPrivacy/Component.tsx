import React, {memo} from 'react';

import {tProps} from './_types';

export const EventPrivacyComponent = memo((props: tProps) => (
  <small className="bgYellowLite br8 p1 pL2 pR2 mR2">
    {props.isPrivate && (
      <>
        <span
          role="img"
          className="mR1"
          aria-label="Lock Emoji">
          ðŸ”’
        </span>
        Private Event
      </>
    )}
    {!props.isPrivate && (
      <>
        <span
          role="img"
          className="mR1"
          aria-label="Tada Emoji">
          ðŸŽ‰
        </span>
        Public Event
      </>
    )}
  </small>
));
