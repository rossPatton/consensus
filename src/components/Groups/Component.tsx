import cx from 'classnames';
import _ from 'lodash';
import React, {memo} from 'react';

import {Group} from './_components';
import {tComponentProps} from './_types';

export const GroupsComponent = memo((props: tComponentProps) => (
  <ul className={cx({'flex flex-wrap': !props.asList})}>
    {props.groups.length > 0
      && props.pendingOrgs.length > 0
      && (
        <li className="font-bold mb-2 pb-2 border-b-1">
          Current group memberships
        </li>
      )}
    {props.groups.map((group, i) => (
      <Group
        {...props}
        key={group.handle}
        group={group}
        index={i}
      />
    ))}
    {props.pendingOrgs.length > 0 && (
      <>
        <li className="font-bold mb-2 pb-2 border-b-1">
          Pending group memberships
        </li>
        {props.pendingOrgs.map((pendingOrg, i) => (
          <Group
            {...props}
            index={i}
            key={pendingOrg.id}
            group={pendingOrg}
          />
        ))}
      </>
    )}
  </ul>
));
