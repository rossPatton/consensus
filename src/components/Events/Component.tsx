import cx from 'classnames';
import dayJS from 'dayjs';
import _ from 'lodash';
import qs from 'query-string';
import React, {memo} from 'react';
import {Link} from 'react-router-dom';

import {ExternalLink, PlaceholderImage, RSVP} from '../../components';
import {objToQueryString} from '../../utils';
import {tComponentProps} from './_types';

export const EventsComponent = memo((props: tComponentProps) => (
  <ul
    className={cx({
      'fx fxdRow taL': props.horizontal,
      'jcBetween': props.horizontal && props.events.length === 4,
    })}>
    {props.events.map((ev, i) => {
      const isPastEvent = dayJS(ev.date).isBefore(dayJS());

      return (
        <li
          key={ev.id}
          className={cx({
            'mB2': !props.horizontal,
            'col row': props.horizontal,
            'mR4': props.horizontal && i !== props.events.length - 1,
          })}>
          <div
            className={cx({
              'fx aiCtr': !props.horizontal,
              'p3 hvrBgGrey1 br8': props.isEditable,
            })}>
            <div
              className={cx({
                'mB2 bgWhite': props.horizontal,
                'mR3 bgGrey2': !props.horizontal,
              })}>
              <Link
                className="noUnderline"
                to={ev.isDraft
                  ? `/org/${ev.orgId}/planMeeting?${qs.stringify(ev)}`
                  : `/event/${ev.id}`}>
                <PlaceholderImage
                  height={100}
                  seed={ev.id}
                  width={200}
                />
              </Link>
            </div>
            <div>
              <div
                className={cx({
                  'fx aiCtr mB1 fs7 fw600 lh1': true,
                  o5: isPastEvent,
                })}>
                <time className="mR1" dateTime={ev.date}>
                  {dayJS(ev.date).format('MMM DD | h:mmA')}
                </time>
                {!props.horizontal && (
                  <>
                    <span className="mR1">@</span>
                    {ev.locationLink && (
                      <ExternalLink
                        noFollow
                        className="mR1"
                        to={ev.locationLink}>
                        {ev.location}
                      </ExternalLink>
                    )}
                  </>
                )}
                {!ev.locationLink && ev.location}
              </div>
              <h3
                className={cx({
                  'fx aiCtr ttCap': true,
                  mB1: props.isEditable,
                  fs4: props.horizontal,
                })}>
                <Link
                  className="noUnderline"
                  to={ev.isDraft
                    ? `/org/${ev.orgId}/planMeeting?${qs.stringify(ev)}`
                    : `/event/${ev.id}`}>
                  {ev.title}
                </Link>
              </h3>
              {props.showRSVPs && (
                <div
                  className={cx({
                    'fx aiCtr fs6 lh1': true,
                    hide: props.horizontal,
                    o5: isPastEvent,
                  })}>
                  {!ev.isDraft && <RSVP event={ev} />}
                  {ev.isDraft && 'This event is not published yet. Click to edit.'}
                </div>
              )}
              {props.showOrgName
                && (
                  <Link
                    to={`/org/${ev.orgId}`}
                    className={cx({
                      'fw600 fs6 lh1': true,
                      o5: isPastEvent,
                    })}>
                    {ev.orgName}
                  </Link>
                )}
              {!isPastEvent
              && props.isEditable
              && (
                <div className="fx aiCtr mB1">
                  <div className="bgGrey4 br4 fs7 lh1 mR2 p1 pL2 pR2 white">
                    {ev.isDraft ? 'Draft' : 'Published'}
                  </div>
                  <div className="fx aiCtr mR1">
                    <Link
                      to={`/org/${ev.orgId}/planMeeting?${objToQueryString(ev)}`}
                      className="btn fs7 fw600 hvrBgGrey1 lh1 noUnderline p1 pL2 pR2">
                      <span
                        role="img"
                        aria-label="Hand with Pen Emoji">
                        ✍️
                      </span>
                      Edit
                    </Link>
                  </div>
                  <button
                    onClick={e => props.deleteEvent(e, ev.id)}
                    className="aiCtr fs7 fw600 fx hvrBgGrey1 lh1">
                    <span
                      role="img"
                      className="mR1"
                      aria-label="Big X Emoji">
                      ✖️
                    </span>
                    Delete
                  </button>
                </div>
              )}
            </div>
          </div>
        </li>
      );
    })}
  </ul>
));
