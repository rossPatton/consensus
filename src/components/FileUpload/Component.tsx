import cx from 'classnames';
import React, {memo} from 'react';
import {v1} from 'uuid';

import {Emoji} from '~app/components';

import {tComponentProps} from './_types';

const baseUrl = 'https://consensus.nyc3.digitaloceanspaces.com';

export const FileUploadComponent = memo((props: tComponentProps) => (
  <>
    <h3 className="text-base mb-1">
      {props.title}
    </h3>
    <div
      id="imagePreview"
      className={cx({
        'animated bg-gray-1 p-3 mb-3 flex flex-col items-center': true,
        'fadeInDown': props.img,
        'hidden': !props.img,
      })}>
      <button
        type="button"
        onClick={props.removeImage}
        className="mb-2 p-2">
        <Emoji
          label="Close Emoji"
          emoji="✖️"
        />
        Remove Image
      </button>
      {props.img && (
        <img
          alt="Your uploaded file preview."
          height={props.width}
          width={props.width}
          src={`${baseUrl}/${props.folder}/${props.img}${__DEV__ ? `?v=${v1()}` : ''}`}
          className={cx({
            'rounded w-full': !props.fieldKey.includes('Avatar'),
            'rounded-circ': props.fieldKey.includes('Avatar'),
          })}
        />
      )}
    </div>
    <label
      htmlFor="fileUpload"
      className={cx({
        'fadeInDown bg-gray-1 border cursor-ptr flex flex-col items-center mb-3 p-3 rounded animated': true,
        'fadeInDown': !props.img,
        'hidden': props.img,
      })}>
      <input
        accept="image/png, image/jpeg, image/gif"
        id="fileUpload"
        name="eventFeaturedImage"
        onChange={props.setImage}
        type="file"
        className={cx({
          hidden: true,
        })}
      />
      <span className="btn flex items-center mb-1 p-1">
        <Emoji
          label="Picture (Camera) Emoji"
          emoji="📷"
        />
        Upload Your Image
      </span>
      {props.info && (
        <small>{props.info}</small>
      )}
    </label>
  </>
));
