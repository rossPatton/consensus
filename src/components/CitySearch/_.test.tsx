import React from 'react';
import { Provider } from 'react-redux';
import render from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import {testGeo, testGroup1, testNYC, testUserSession} from '~app/constants/jest';

import CitySearch from '.';
import {CitySearchComponent} from './Component';

const mockStore = configureStore([thunk]);
const defaultStore = {
  cities: {
    error: null as Error | null,
    isLoading: false,
    data: [testNYC],
  },
  geo: {
    error: null as Error | null,
    isLoading: false,
    data: testGeo,
  },
  session: testUserSession,
};

describe('components/CitySearch', () => {
  let store = mockStore({});

  beforeEach(() => {
    store = mockStore(defaultStore);
  });

  it('renders without crashing', () => {
    const component = render.create((
      <Provider store={store}>
        <CitySearch
          city="New York"
          cityId={16625}
          // getCitiesDispatch={jest.fn()}
          region="New York"
          regionId={37}
          updateState={jest.fn()}
        />
      </Provider>
    ));
    expect(component).toMatchSnapshot();
  });

  it('renders component without crashing', () => {
    const component = render.create((
      <CitySearchComponent
        city="New York"
        cityId={16625}
        cities={[{
          country: 'United States',
          countryId: 1,
          groups: [testGroup1],
          name: 'New York',
          id: 16625,
          postcodes: [10002],
          region: 'New York',
          regionId: 37,
        }]}
        geo={testGeo}
        region="New York"
        regionId={37}
        // @ts-ignore
        session={defaultStore.session}
        showRemoveButton
        updateState={jest.fn()}
      />
    ));
    expect(component).toMatchSnapshot();
  });

  it('renders component with blank city and region', () => {
    const component = render.create((
      <CitySearchComponent
        city=""
        cityId={0}
        cities={[{
          country: 'United States',
          countryId: 1,
          groups: [testGroup1],
          name: 'New York',
          id: 16625,
          postcodes: [10002],
          region: 'New York',
          regionId: 37,
        }]}
        geo={testGeo}
        region=""
        regionId={0}
        // @ts-ignore
        session={defaultStore.session}
        updateState={jest.fn()}
      />
    ));
    expect(component).toMatchSnapshot();
  });

  it('renders with custom label', () => {
    const component = render.create((
      <Provider store={store}>
        <CitySearch
          city="New York"
          cityId={16625}
          label="Testing custom label"
          region="New York"
          regionId={37}
          updateState={jest.fn()}
        />
      </Provider>
    ));
    expect(component).toMatchSnapshot();
  });

  it('renders with remove button', () => {
    const component = render.create((
      <Provider store={store}>
        <CitySearch
          city="New York"
          cityId={16625}
          region="New York"
          regionId={37}
          showRemoveButton
          updateState={jest.fn()}
        />
      </Provider>
    ));
    expect(component).toMatchSnapshot();
  });

  it('renders with blank session region', () => {
    const customStore = {
      ...store,
      session: {
        error: null as Error | null,
        isLoading: false,
        data: {
          isAuthenticated: false,
          profile: {
            region: '',
          },
        },
      },
    };
    const component = render.create((
      <Provider store={customStore}>
        <CitySearch
          city="New York"
          cityId={16625}
          region="New York"
          regionId={37}
          showRemoveButton
          updateState={jest.fn()}
        />
      </Provider>
    ));
    expect(component).toMatchSnapshot();
  });

  it('renders with null session region', () => {
    const customStore = {
      ...store,
      session: {
        error: null as Error | null,
        isLoading: false,
        data: {
          isAuthenticated: false,
          profile: {
            region: null as string | null,
          },
        },
      },
    };
    const component = render.create((
      <Provider store={customStore}>
        <CitySearch
          city="New York"
          cityId={16625}
          region="New York"
          regionId={37}
          showRemoveButton
          updateState={jest.fn()}
        />
      </Provider>
    ));
    expect(component).toMatchSnapshot();
  });
});
