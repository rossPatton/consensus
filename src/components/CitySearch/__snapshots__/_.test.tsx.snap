// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CitySearch renders component with blank city and region 1`] = `
Array [
  <div
    className=""
  >
    <h2
      className="text-base mb-1"
    >
      Where are you located?
    </h2>
    <label
      htmlFor="stateSelect"
    >
      <select
        className="mb-3 w-full"
        id="stateSelect"
        name="type"
        onBlur={[Function]}
        onChange={[Function]}
        value=""
      >
        <option
          value=""
        >
          State Select
        </option>
        <option
          value="Alabama"
        >
          Alabama
        </option>
        <option
          value="Alaska"
        >
          Alaska
        </option>
        <option
          value="American Samoa"
        >
          American Samoa
        </option>
        <option
          value="Arizona"
        >
          Arizona
        </option>
        <option
          value="Arkansas"
        >
          Arkansas
        </option>
        <option
          value="California"
        >
          California
        </option>
        <option
          value="Colorado"
        >
          Colorado
        </option>
        <option
          value="Connecticut"
        >
          Connecticut
        </option>
        <option
          value="Delaware"
        >
          Delaware
        </option>
        <option
          value="District Of Columbia"
        >
          District Of Columbia
        </option>
        <option
          value="Federated States Of Micronesia"
        >
          Federated States Of Micronesia
        </option>
        <option
          value="Florida"
        >
          Florida
        </option>
        <option
          value="Georgia"
        >
          Georgia
        </option>
        <option
          value="Guam"
        >
          Guam
        </option>
        <option
          value="Hawaii"
        >
          Hawaii
        </option>
        <option
          value="Idaho"
        >
          Idaho
        </option>
        <option
          value="Illinois"
        >
          Illinois
        </option>
        <option
          value="Indiana"
        >
          Indiana
        </option>
        <option
          value="Iowa"
        >
          Iowa
        </option>
        <option
          value="Kansas"
        >
          Kansas
        </option>
        <option
          value="Kentucky"
        >
          Kentucky
        </option>
        <option
          value="Louisiana"
        >
          Louisiana
        </option>
        <option
          value="Maine"
        >
          Maine
        </option>
        <option
          value="Marshall Islands"
        >
          Marshall Islands
        </option>
        <option
          value="Maryland"
        >
          Maryland
        </option>
        <option
          value="Massachusetts"
        >
          Massachusetts
        </option>
        <option
          value="Michigan"
        >
          Michigan
        </option>
        <option
          value="Minnesota"
        >
          Minnesota
        </option>
        <option
          value="Mississippi"
        >
          Mississippi
        </option>
        <option
          value="Missouri"
        >
          Missouri
        </option>
        <option
          value="Montana"
        >
          Montana
        </option>
        <option
          value="Nebraska"
        >
          Nebraska
        </option>
        <option
          value="Nevada"
        >
          Nevada
        </option>
        <option
          value="New Hampshire"
        >
          New Hampshire
        </option>
        <option
          value="New Jersey"
        >
          New Jersey
        </option>
        <option
          value="New Mexico"
        >
          New Mexico
        </option>
        <option
          value="New York"
        >
          New York
        </option>
        <option
          value="North Carolina"
        >
          North Carolina
        </option>
        <option
          value="North Dakota"
        >
          North Dakota
        </option>
        <option
          value="Northern Mariana Islands"
        >
          Northern Mariana Islands
        </option>
        <option
          value="Ohio"
        >
          Ohio
        </option>
        <option
          value="Oklahoma"
        >
          Oklahoma
        </option>
        <option
          value="Oregon"
        >
          Oregon
        </option>
        <option
          value="Palau"
        >
          Palau
        </option>
        <option
          value="Pennsylvania"
        >
          Pennsylvania
        </option>
        <option
          value="Puerto Rico"
        >
          Puerto Rico
        </option>
        <option
          value="Rhode Island"
        >
          Rhode Island
        </option>
        <option
          value="South Carolina"
        >
          South Carolina
        </option>
        <option
          value="South Dakota"
        >
          South Dakota
        </option>
        <option
          value="Tennessee"
        >
          Tennessee
        </option>
        <option
          value="Texas"
        >
          Texas
        </option>
        <option
          value="Utah"
        >
          Utah
        </option>
        <option
          value="Vermont"
        >
          Vermont
        </option>
        <option
          value="Virgin Islands"
        >
          Virgin Islands
        </option>
        <option
          value="Virginia"
        >
          Virginia
        </option>
        <option
          value="Washington"
        >
          Washington
        </option>
        <option
          value="West Virginia"
        >
          West Virginia
        </option>
        <option
          value="Wisconsin"
        >
          Wisconsin
        </option>
        <option
          value="Wyoming"
        >
          Wyoming
        </option>
      </select>
    </label>
  </div>,
  "",
  <div
    className="hidden mb-3"
  >
    <label
      htmlFor="citySelect"
    >
      <select
        className="w-full"
        id="citySelect"
        name="type"
        onBlur={[Function]}
        onChange={[Function]}
        value=""
      >
        <option
          value="New York"
        >
          New York
          , 
          New York
        </option>
      </select>
    </label>
  </div>,
]
`;

exports[`components/CitySearch renders component without crashing 1`] = `
Array [
  <div
    className="hidden"
  >
    <h2
      className="text-base mb-1"
    >
      Where are you located?
    </h2>
    <label
      htmlFor="stateSelect"
    >
      <select
        className="mb-3 w-full"
        id="stateSelect"
        name="type"
        onBlur={[Function]}
        onChange={[Function]}
        value="New York"
      >
        <option
          value=""
        >
          State Select
        </option>
        <option
          value="Alabama"
        >
          Alabama
        </option>
        <option
          value="Alaska"
        >
          Alaska
        </option>
        <option
          value="American Samoa"
        >
          American Samoa
        </option>
        <option
          value="Arizona"
        >
          Arizona
        </option>
        <option
          value="Arkansas"
        >
          Arkansas
        </option>
        <option
          value="California"
        >
          California
        </option>
        <option
          value="Colorado"
        >
          Colorado
        </option>
        <option
          value="Connecticut"
        >
          Connecticut
        </option>
        <option
          value="Delaware"
        >
          Delaware
        </option>
        <option
          value="District Of Columbia"
        >
          District Of Columbia
        </option>
        <option
          value="Federated States Of Micronesia"
        >
          Federated States Of Micronesia
        </option>
        <option
          value="Florida"
        >
          Florida
        </option>
        <option
          value="Georgia"
        >
          Georgia
        </option>
        <option
          value="Guam"
        >
          Guam
        </option>
        <option
          value="Hawaii"
        >
          Hawaii
        </option>
        <option
          value="Idaho"
        >
          Idaho
        </option>
        <option
          value="Illinois"
        >
          Illinois
        </option>
        <option
          value="Indiana"
        >
          Indiana
        </option>
        <option
          value="Iowa"
        >
          Iowa
        </option>
        <option
          value="Kansas"
        >
          Kansas
        </option>
        <option
          value="Kentucky"
        >
          Kentucky
        </option>
        <option
          value="Louisiana"
        >
          Louisiana
        </option>
        <option
          value="Maine"
        >
          Maine
        </option>
        <option
          value="Marshall Islands"
        >
          Marshall Islands
        </option>
        <option
          value="Maryland"
        >
          Maryland
        </option>
        <option
          value="Massachusetts"
        >
          Massachusetts
        </option>
        <option
          value="Michigan"
        >
          Michigan
        </option>
        <option
          value="Minnesota"
        >
          Minnesota
        </option>
        <option
          value="Mississippi"
        >
          Mississippi
        </option>
        <option
          value="Missouri"
        >
          Missouri
        </option>
        <option
          value="Montana"
        >
          Montana
        </option>
        <option
          value="Nebraska"
        >
          Nebraska
        </option>
        <option
          value="Nevada"
        >
          Nevada
        </option>
        <option
          value="New Hampshire"
        >
          New Hampshire
        </option>
        <option
          value="New Jersey"
        >
          New Jersey
        </option>
        <option
          value="New Mexico"
        >
          New Mexico
        </option>
        <option
          value="New York"
        >
          New York
        </option>
        <option
          value="North Carolina"
        >
          North Carolina
        </option>
        <option
          value="North Dakota"
        >
          North Dakota
        </option>
        <option
          value="Northern Mariana Islands"
        >
          Northern Mariana Islands
        </option>
        <option
          value="Ohio"
        >
          Ohio
        </option>
        <option
          value="Oklahoma"
        >
          Oklahoma
        </option>
        <option
          value="Oregon"
        >
          Oregon
        </option>
        <option
          value="Palau"
        >
          Palau
        </option>
        <option
          value="Pennsylvania"
        >
          Pennsylvania
        </option>
        <option
          value="Puerto Rico"
        >
          Puerto Rico
        </option>
        <option
          value="Rhode Island"
        >
          Rhode Island
        </option>
        <option
          value="South Carolina"
        >
          South Carolina
        </option>
        <option
          value="South Dakota"
        >
          South Dakota
        </option>
        <option
          value="Tennessee"
        >
          Tennessee
        </option>
        <option
          value="Texas"
        >
          Texas
        </option>
        <option
          value="Utah"
        >
          Utah
        </option>
        <option
          value="Vermont"
        >
          Vermont
        </option>
        <option
          value="Virgin Islands"
        >
          Virgin Islands
        </option>
        <option
          value="Virginia"
        >
          Virginia
        </option>
        <option
          value="Washington"
        >
          Washington
        </option>
        <option
          value="West Virginia"
        >
          West Virginia
        </option>
        <option
          value="Wisconsin"
        >
          Wisconsin
        </option>
        <option
          value="Wyoming"
        >
          Wyoming
        </option>
      </select>
    </label>
  </div>,
  <h2
    className="text-base mb-2"
  >
    City in 
    <span
      className="inline-block mr-2"
    >
      New York
    </span>
    <button
      className="p-1 mr-1"
      onClick={[Function]}
      type="button"
    >
      Not the right state?
    </button>
    <button
      className="p-1"
      onClick={[Function]}
      type="button"
    >
      Reset City Search
    </button>
  </h2>,
  <div
    className="mb-3"
  >
    <div
      className="border p-2 rounded inline-block"
    >
      <b>
        New York
      </b>
      , 
      <span
        className="inline-block mr-3"
      >
        New York
      </span>
      <button
        className="p-1"
        onClick={[Function]}
        type="button"
      >
        Not the right city?
      </button>
    </div>
  </div>,
]
`;

exports[`components/CitySearch renders with blank session region 1`] = `null`;

exports[`components/CitySearch renders with custom label 1`] = `null`;

exports[`components/CitySearch renders with null session region 1`] = `null`;

exports[`components/CitySearch renders with remove button 1`] = `null`;

exports[`components/CitySearch renders without crashing 1`] = `null`;
